### PR TITLE
Adjust shuffle tests for dask-expr

### PR DESCRIPTION
**Pull request by phofl**
- [ ] Closes #xxxx
- [ ] Tests added / passed
- [ ] Passes `pre-commit run --all-files`

### CODE REVIEW

File: dask/dataframe/tests/test_shuffle.py [hunk 1]
```diff
@@ -583,6 +613,8 @@ def test_npartitions_auto_raises_deprecation_warning():
 
 
 def test_set_index_doesnt_increase_partitions(shuffle_method):
+    if DASK_EXPR_ENABLED:
+        pytest.skip(reason="auto not supported")
```
Comment (crusaderky) [inline dask/dataframe/tests/test_shuffle.py:617]:
```suggestion
@pytest.mark.skipif(DASK_EXPR_ENABLED, reason="auto not supported")
def test_set_index_doesnt_increase_partitions(shuffle_method):
```

File: dask/dataframe/tests/test_shuffle.py [hunk 2]
```diff
@@ -819,6 +862,8 @@ def test_set_index_interpolate_large_uint(engine):
 
 
 def test_set_index_timezone():
+    if DASK_EXPR_ENABLED:
+        pytest.skip(reason="we don't do division inference for 1 partition frames")
```
Comment (crusaderky) [inline dask/dataframe/tests/test_shuffle.py:866]:
```suggestion
@pytest.mark.skipif(DASK_EXPR_ENABLED, reason="we don't do division inference for 1 partition frames")
def test_set_index_timezone():
```

File: dask/dataframe/tests/test_shuffle.py [hunk 3]
```diff
@@ -952,6 +1001,8 @@ def test_set_index_sorted_single_partition():
 
 
 def test_set_index_sorted_min_max_same():
+    if DASK_EXPR_ENABLED:
+        pytest.skip(reason="we don't do division inference for sorted=True")
```
Comment (crusaderky) [inline dask/dataframe/tests/test_shuffle.py:1005]:
```suggestion
@pytest.mark.skipif(DASK_EXPR_ENABLED, reason="we don't do division inference for sorted=True")
def test_set_index_sorted_min_max_same():
```

File: dask/dataframe/tests/test_shuffle.py [hunk 4]
```diff
@@ -1114,6 +1173,8 @@ def make_part(dummy, n):
 
 
 def test_set_index_errors_with_inplace_kwarg():
+    if DASK_EXPR_ENABLED:
+        pytest.skip(reason="we don't support inplace there")
```
Comment (crusaderky) [inline dask/dataframe/tests/test_shuffle.py:1177]:
```suggestion
@pytest.mark.skipif(DASK_EXPR_ENABLED, reason="inplace not supported")
def test_set_index_errors_with_inplace_kwarg():
```

File: dask/dataframe/tests/test_shuffle.py [hunk 5]
```diff
@@ -1091,6 +1148,8 @@ def test_gh_2730():
 
 
 def test_set_index_does_not_repeat_work_due_to_optimizations():
+    if DASK_EXPR_ENABLED:
+        pytest.skip(reason="we test this over in dask-expr")
```
Comment (crusaderky) [inline dask/dataframe/tests/test_shuffle.py:1152]:
```suggestion
@pytest.mark.skipif(DASK_EXPR_ENABLED, reason="we test this over in dask-expr")
def test_set_index_does_not_repeat_work_due_to_optimizations():
```

File: dask/dataframe/tests/test_shuffle.py [hunk 6]
```diff
@@ -1312,6 +1372,8 @@ def test_compute_current_divisions_overlap_2():
 
 
 def test_shuffle_hlg_layer():
+    if DASK_EXPR_ENABLED:
+        pytest.skip("not valid for dask-expr")
```
Comment (crusaderky) [inline dask/dataframe/tests/test_shuffle.py:1376]:
```suggestion
@pytest.mark.skipif(DASK_EXPR_ENABLED, reason="not valid for dask-expr")
def test_shuffle_hlg_layer():
```

File: dask/dataframe/tests/test_shuffle.py [hunk 7]
```diff
@@ -1374,6 +1439,8 @@ def test_shuffle_partitions_meta_dtype():
     ],
 )
 def test_shuffle_hlg_layer_serialize(npartitions):
+    if DASK_EXPR_ENABLED:
+        pytest.skip("not valid for dask-expr")
```
Comment (crusaderky) [inline dask/dataframe/tests/test_shuffle.py:1443]:
```suggestion
@pytest.mark.skipif(DASK_EXPR_ENABLED, reason="not valid for dask-expr")
def test_shuffle_hlg_layer_serialize(npartitions):
```

File: dask/dataframe/tests/test_shuffle.py [hunk 8]
```diff
@@ -1458,6 +1525,8 @@ def test_set_index_with_series_uses_fastpath():
 
 
 def test_set_index_partitions_meta_dtype():
+    if DASK_EXPR_ENABLED:
+        pytest.skip("not valid for dask-expr")
```
Comment (crusaderky) [inline dask/dataframe/tests/test_shuffle.py:1529]:
```suggestion
@pytest.mark.skipif(DASK_EXPR_ENABLED, reason="not valid for dask-expr")
def test_set_index_partitions_meta_dtype():
```

File: dask/dataframe/tests/test_shuffle.py [hunk 9]
```diff
@@ -1473,6 +1542,8 @@ def test_set_index_partitions_meta_dtype():
 
 
 def test_sort_values_partitions_meta_dtype_with_divisions():
+    if DASK_EXPR_ENABLED:
+        pytest.skip("not valid for dask-expr")
```
Comment (crusaderky) [inline dask/dataframe/tests/test_shuffle.py:1546]:
```suggestion
@pytest.mark.skipif(DASK_EXPR_ENABLED, reason="not valid for dask-expr")
def test_sort_values_partitions_meta_dtype_with_divisions():
```

**Review by crusaderky — CHANGES_REQUESTED** [review]
The one non-cosmetic change is support for duplicated columns changed from skip to xfail. It's a missing feature in dask-expr we expect to add eventually.

File: dask/dataframe/__init__.py [hunk 10]
```diff
@@ -33,11 +33,25 @@ def _dask_expr_enabled() -> bool:
         merge,
         pivot_table,
         read_csv,
+        read_hdf,
```
Comment (crusaderky) [inline dask/dataframe/__init__.py:36]: tests are failing:

FAILED dask/array/tests/test_array_core.py::test_map_blocks_series - ImportError: cannot import name 'read_hdf' from 'dask_expr' (/home/runner/miniconda3/envs/test-environment/lib/python3.12/site-packages/dask_expr/__init__.py). Did you mean: 'read_csv'?
FAILED dask/tests/test_distributed.py::test_to_hdf_distributed - ImportError: cannot import name 'read_hdf' from 'dask_expr' (/home/runner/miniconda3/envs/test-environment/lib/python3.12/site-packages/dask_expr/__init__.py). Did you mean: 'read_csv'?
FAILED dask/tests/test_distributed.py::test_to_hdf_scheduler_distributed[1] - ImportError: cannot import name 'read_hdf' from 'dask_expr' (/home/runner/miniconda3/envs/test-environment/lib/python3.12/site-packages/dask_expr/__init__.py). Did you mean: 'read_csv'?
FAILED dask/tests/test_order.py::test_array_vs_dataframe[True] - ImportError: cannot import name 'read_hdf' from 'dask_expr' (/home/runner/miniconda3/envs/test-environment/lib/python3.12/site-packages/dask_expr/__init__.py). Did you mean: 'read_csv'?
FAILED dask/tests/test_order.py::test_array_vs_dataframe[False] - ImportError: cannot import name 'read_hdf' from 'dask_expr' (/home/runner/miniconda3/envs/test-environment/lib/python3.12/site-packages/dask_expr/__init__.py). Did you mean: 'read_csv'?

ERROR collecting dask/dataframe/io/tests/test_demo.py
  E   ImportError: cannot import name 'read_hdf' from 'dask_expr' (/home/runner/miniconda3/envs/test-environment/lib/python3.12/site-packages/dask_expr/__init__.py). Did you mean: 'read_csv'?
```

Comment (phofl) [reply dask/dataframe/__init__.py:36]: This must be because of cache issues, I've merged all he io connectors yesterday evening in dask-expr

**Comment by phofl** [comment]
The rerun looks better

File: dask/dataframe/tests/test_dataframe.py [hunk 11]
```diff
@@ -5714,6 +5722,8 @@ def test_dataframe_groupby_cumprod_agg_empty_partitions():
 
 
 def test_fuse_roots():
+    if DASK_EXPR_ENABLED:
+        pytest.skip("doesn't nmake sense")
```
[the remark this answers is not in the document]
Comment (crusaderky) [reply dask/dataframe/tests/test_dataframe.py:5725]: This is so weird. I'm investigating but I won't block the PR for it.

**Review by crusaderky — APPROVED** [review]

File: dask/dataframe/tests/test_dataframe.py (hunk 11 above)
[the remark this answers is not in the document]
Comment (phofl) [reply dask/dataframe/tests/test_dataframe.py:5725]: lets not spend too much time on this, it's not really a problem for the moment

[the remark this answers is not in the document]
Comment (crusaderky) [reply dask/dataframe/tests/test_dataframe.py:5725]: The reason was a missing $ sign on line 92 here: https://github.com/dask/dask/pull/10766/files
This in turn caused the config variable to become a literal string "DASK_DATAFRAME__CONVERT_STRING", which in a if statement evaluates as True, but in `pytest.mark.skipif` triggers an eval() call.

[the remark this answers is not in the document]
Comment (phofl) [reply dask/dataframe/tests/test_dataframe.py:5725]: Yikes, I'll create a PR moving them back to parametric

[the remark this answers is not in the document]
Comment (crusaderky) [reply dask/dataframe/tests/test_dataframe.py:5725]: https://github.com/dask/dask/pull/10774

**Pull request merged**